receiver/pulsarreceiver/pulsar_receiver.go

```diff
@@ -81,13 +94,16 @@ func consumerTracesLoop(ctx context.Context, c *pulsarTracesConsumer) error {
 
 	// TODO: Ensure returned errors are handled
 	for {
+		obsCtx := c.obsrecv.StartTracesOp(ctx)
 		message, err := c.consumer.Receive(ctx)
 		if err != nil {
 			if strings.Contains(err.Error(), alreadyClosedError) {
+				c.obsrecv.EndTracesOp(obsCtx, unmarshaler.Encoding(), 0, err)
 				return err
 			}
 			if errors.Is(err, context.Canceled) {
 				c.settings.Logger.Info("exiting consume traces loop")
+				c.obsrecv.EndTracesOp(obsCtx, unmarshaler.Encoding(), 0, err)
```

Comment:
Shouldn't this be moved under `if err != nil {` so we close the trace on any failure?

Reply:
Oh yeah. That's actually better.

Reply:
Done. PTAL!